src/server/game/Handlers/MiscHandler.cpp

```diff
@@ -98,6 +98,12 @@ void WorldSession::HandleGossipSelectOptionOpcode(WorldPacket& recv_data)
     if (_player->PlayerTalkClass->IsGossipOptionCoded(gossipListId))
         recv_data >> code;
 
+    // Prevent cheating on C++ scripted menus
+    if (_player->PlayerTalkClass->GetGossipMenu().GetSenderGUID() != guid)
```

Comment:
would this not break alot of exsisting custom npc gossips that are not based on guid of the character due to them being the need to be manaully spawned?

Reply:
No. It's not a player's guid, but the guid of NPC you are talking to. You cannot select any option from other NPCs that you are not talking to.

Reply:
> No. It's not a player's guid, but the guid of NPC you are talking to. You cannot select any option from other NPCs that you are not talking to.

I apologize, that is what i ment, the guid of the npc is not matching the gossip menu id. We have a few custom gossip npcs. Transmog npc being one wher we have to manually add to the playing field which will then give it a different guid from it's gossip.

Reply:
`I apologize, that is what i ment, the guid of the npc is not matching the gossip menu id` Then it's wrong.